File: 2018/haskell/benchmark/Bench.hs
import           Criterion.Main                 ( bench
                                                , bgroup
                                                , defaultMain
                                                , nf
                                                , nfIO
                                                , whnf
                                                )
import           Day01                          ( day01a
                                                , day01b
                                                )

runBench :: String -> ([String] -> Maybe Int) -> IO ()
runBench name f = do
  x <- readFile $ "../data/" ++ name ++ ".txt"
  let _ = f $ lines x
  return ()

main :: IO ()
main = do
  defaultMain
    [ bgroup
        "adv"
        [ bench "1a" $ nfIO $ runBench "day01" day01a
        , bench "1b" $ nfIO $ runBench "day01" day01b
        ]
    ]

File: 2018/rust/benches/bench.rs
#[macro_use]
extern crate criterion;
extern crate advent_of_code_2018;
use advent_of_code_2018::utils::*;
use advent_of_code_2018::*;
use criterion::{black_box, Criterion};
use pprof::criterion::{Output, PProfProfiler};

fn bench_fi(c: &mut Criterion) {
    let day01 = &read("day01.txt");
    c.bench_function("day01a", |b| b.iter(|| day01::bench(black_box(day01))));
    // c.bench_function("day01b", |b| b.iter(|| day01::part2(black_box(day01))));
}

criterion_group! {
    name = benches;
    config = Criterion::default().with_profiler(PProfProfiler::new(100, Output::Protobuf));
    targets = bench_fi
}

criterion_main!(benches);
